### PR TITLE
Android release build FIXER

### LINKED ISSUE #1286: Android release build fails
## What went wrong? 😲
 
The logging changes cause an issue in the release build

`Permission to access file: /storage/emulated/0/Documents/omSupply/temp.log is denied`

Building is also unhappy about the deprecated method call

## Expected behaviour 🤔
Should run ok and log ok

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Try to build and run a release build

## Your environment 🌱
 - Version [e.g. 1.2.3]
 - Platform [e.g. tablet, browser, windows desktop, macOS desktop, server]

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1286 

# 👩🏻‍💻 What does this PR do? 
 
I could not find a way to safely log to a shared location on android so have reverted those changes and we now log to the app directory.

In order to catch errors which would prevent the UI from being shown, and therefore the logs being accessed, the logger will also log to logcat ( thanks @clemens-msupply! )

Have also reverted the change to put `logging_init` into `server/lib` because that caused an issue when running on windows 🙄 ( the windows service instantiates the logger, due to the difference in fetching settings when run as a service ).

# 🧪 How has/should this change been tested? 

Have built both debug and release builds, run and checked local logs and logcat entries.
Ran the server on mac and checked compilation and logging (features = postgres). Ran `cargo test` to check sqlite build.

## 📃 Documentation

_No user facing changes_

### CODE REVIEW

File: server/server/src/logging.rs [hunk 1]
```diff
@@ -18,8 +19,32 @@ pub fn logging_init(settings: Option<LoggingSettings>) {
...
     };
-    fast_log::init(config.level(LevelFilter::from(settings.level.clone())))
-        .expect("Unable to initialise logger");
+    fast_log::init(
+        config
+            .level(LevelFilter::from(settings.level.clone()))
+            .custom(NativeLogger {}),
+    )
+    .expect("Unable to initialise logger");
+    log::warn!("omSupply Logging initialised aaabbbccc");
```
Comment (clemens-msupply) [inline server/server/src/logging.rs:28]: Remove this ? or fix message :)

File: server/server/src/logging.rs [hunk 2]
```diff
@@ -18,8 +19,32 @@ pub fn logging_init(settings: Option<LoggingSettings>) {
...
+}
+
+struct NativeLogger {}
+impl LogAppender for NativeLogger {
+    fn do_logs(&self, records: &[FastLogRecord]) {
```
Comment (clemens-msupply) [inline server/server/src/logging.rs:33]: You could have a logging_init signature like:

```rust
pub fn logging_init<Appender: LogAppender + 'static>(settings: Option<LoggingSettings>, arg: Option<Appender>)
```

and then pass in the Android logcat logger in the android crate, i.e. no android dependencies in the server crate. (Untested so there might be some pitfalls)

**Review by clemens-msupply — APPROVED** [review]
Looks good to me! thank you

One possible improvement on how to keep the android dependency out of the server crate in the comments...

File: server/server/src/logging.rs (hunk 1 above)
Comment (mark-prins) [reply server/server/src/logging.rs:28]: oops! 🙏

File: server/server/Cargo.toml [hunk 3]
```diff
@@ -51,6 +51,7 @@ futures = "0.3"
 fast_log = { version = "1.5" }
 rcgen = "0.9.2"
 
+
```
Comment (clemens-msupply) [inline server/server/Cargo.toml:54]: Nit: extra line here...

**Review by clemens-msupply — APPROVED** [review]
Nice looks good!

File: server/server/Cargo.toml (hunk 3 above)
Comment (mark-prins) [reply server/server/Cargo.toml:54]: oops, saw this too late. have removed, will pop in the next PR

**Pull request merged**